### PR TITLE
Some fixes for `require-tothrow-message`

### PR DESCRIPTION
**Pull request by garyking**
I allow for any argument type, which would include `TemplateLiteral`, `CallExpression`, etc.

I adjusted the tests by throwing actual errors, adding comments, adding more tests. 

I couldn't add a test for `TemplateLiteral` since it requires `babel-eslint`. 

I couldn't break up the commit into multiple commits because one of the precommit hooks staged and commited the entire file, even when I staged only a few lines at a time.

### CODE REVIEW

**Comment by SimenB** [comment]
>I couldn't add a test for TemplateLiteral since it requires babel-eslint

That doesn't make any sense. Would it be enough to bump `ecmaVersion`?

>I couldn't break up the commit into multiple commits because one of the precommit hooks staged and commited the entire file, even when I staged only a few lines at a time.

What I usually do is just do `--no-verify` to skip the hook. Hopefully it'll be better in the future: https://github.com/okonet/lint-staged/pull/75

File: rules/__tests__/require-tothrow-message.js [hunk 1]
```diff
@@ -7,23 +7,44 @@ const ruleTester = new RuleTester();
...
+    // String
+    "expect(function() { throw new Error('a'); }).toThrow('a');",
+    "expect(function() { throw new Error('a'); }).toThrowError('a');",
+
+    // Template literal
+    // {
+    //   code:
+    //     "const a = 'a'; expect(() => { throw new Error('a'); })" +
+    //     '.toThrow(`${a}`);',
+    //   parser: 'babel-eslint',
```
Comment (SimenB) [inline rules/__tests__/require-tothrow-message.js]: I think it should work if you set `ecmaVersion`: https://eslint.org/docs/user-guide/configuring#specifying-parser-options

**Comment by garyking** [comment]
@SimenB 	Okay changes have been made.

**Pull request merged**